rofl-client/ts/README.md

```diff
@@ -0,0 +1,107 @@
+# @oasisprotocol/rofl-client
+
+TypeScript/Node client SDK for **Oasis ROFL**.
+
+## Install
+
+```bash
+npm install @oasisprotocol/rofl-client
+# or
+yarn add @oasisprotocol/rofl-client
+```
+
+## Quickstart
+
+```ts
+import {RoflClient, KeyKind} from '@oasisprotocol/rofl-client';
+
+const client = new RoflClient(); // UDS: /run/rofl-appd.sock
+
+const key = await client.generateKey('my-key');                   // => hex string
+const ed  = await client.generateKey('my-ed25519', KeyKind.ED25519);
+
+await client.setMetadata({key_fingerprint: key.slice(0, 16)});
+const metadata = await client.getMetadata();
+
+const appId = await client.getAppId(); // bech32 string (helper)
+
+// Sign & submit an authenticated transaction (ETH-style)
+const callResultBytes = await client.signAndSubmit({
+  kind: 'eth',
+  gas_limit: 200_000,
+  to: '',                    // empty => contract creation
+  value: 0,
+  data: '0x',                // hex calldata (0x optional)
+});
+// `callResultBytes` is the raw CBOR-encoded CallResult (Uint8Array)
+```
+
+`generateKey` returns the raw private key encoded as a
+**hex string (no `0x` prefix)**.
+
+### RoflClient(options?)
+
+- `url?: string`
+  - `''` (default): UDS at `/run/rofl-appd.sock`
+  - `http(s)://...`: HTTP(S) base URL
+  - `'/path/to.sock'`: custom UDS path
+- `timeoutMs?: number` (default **60000**)
+  - This is a **socket inactivity timeout** (i.e., triggers on no activity).
+  It does not strictly bound total wall-clock time for very slow responses.
+- `userAgent?: string` (default `@oasisprotocol/rofl-client/<version>`)
+  - Override the `User-Agent` header if you need custom telemetry.
+
+### Methods
+
+- `generateKey(keyId: string, kind?: KeyKind): Promise<string>`
+- `getMetadata(): Promise<Record<string, string>>`
+- `setMetadata(metadata: Record<string, string>): Promise<void>`
+- `getAppId(): Promise<string>` (helper)
```

Comment:
Would be great to also add support for the recently added queries endpoint: https://github.com/oasisprotocol/oasis-sdk/pull/2367

Ideally, the client method would use the types for the arguments and responsens https://github.com/oasisprotocol/oasis-sdk/blob/main/client-sdk/ts-web/rt/src/types.ts but we can leave that as a followup if it's to much work at the moment.